source/Mlos.Python/mlos/OptimizerEvaluationTools/OptimizerEvaluatorConfigStore.py

```diff
@@ -0,0 +1,53 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+from mlos.Spaces import CategoricalDimension, DiscreteDimension, Point, SimpleHypergrid
+from mlos.Spaces.Configs import ComponentConfigStore
+
+optimizer_evaluator_config_store = ComponentConfigStore(
+    description="Describes the configuration space for the OptimizerEvaluator.\n"
+                "\n"
+                "Dimensions:\n"
+                "- num_iterations: how many optimization iterations to run.\n"
+                "- evaluation_frequency: how often should the evaluator capture the optima and goodness of fit metrics (e.g. every 10 iterations).\n"
+                "- include_pickled_optimizer_in_report: should the state of the optimizer be pickled and saved.\n"
+                "- include_pickled_objective_function_in_report: should the final state of the objective function be pickled and saved.\n"
+                "- report_regression_model_goodness_of_fit: should the goodness of fit metrics be included in the evaluation report.\n"
+                "- report_optima_over_time: should the optima over time be included in the evaluation report.\n"
+                "- include_execution_trace_in_report: should the execution trace produced by mlos.Tracer be included in the evaluation report.",
+
+    parameter_space=SimpleHypergrid(
+        name="optimizer_evaluator",
+        dimensions=[
+            DiscreteDimension(name="num_iterations", min=1, max=2**32),
+            DiscreteDimension(name="evaluation_frequency", min=1, max=2**10),
+            CategoricalDimension(name="include_pickled_optimizer_in_report", values=[True, False]),
+            CategoricalDimension(name="include_pickled_objective_function_in_report", values=[True, False]),
+            CategoricalDimension(name="report_regression_model_goodness_of_fit", values=[True, False]),
+            CategoricalDimension(name="report_optima_over_time", values=[True, False]),
+            CategoricalDimension(name="include_execution_trace_in_report", values=[True, False]),
+        ]
+    ),
+    default=Point(
+        num_iterations=100,
+        evaluation_frequency=10,
+        include_pickled_optimizer_in_report=True,
+        include_pickled_objective_function_in_report=True,
+        report_regression_model_goodness_of_fit=True,
+        report_optima_over_time=True,
+        include_execution_trace_in_report=True,
+    )
+)
+
+
+# Parallel unit tests config.

```

Comment:
I generally prefer not to have testing code together with the main code but I guess that's a matter of taste.

Reply:
#163 

I created an issue to solicit opinions from the rest of the team. I think I could go either way. What are your arguments for and against?